src/openrct2/interface/Screenshot.cpp

```diff
@@ -337,23 +277,42 @@ static void ReleaseDPI(rct_drawpixelinfo& dpi)
     dpi.height = 0;
 }
 
-static rct_viewport GetGiantViewport(const TileCoordsXY& mapSize, int32_t rotation, ZoomLevel zoom)
+static rct_viewport GetGiantViewport(int32_t rotation, ZoomLevel zoom)
 {
     // Get the tile coordinates of each corner
-    auto leftTileCoords = GetEdgeTile(mapSize, rotation, EdgeType::LEFT, false);
-    auto rightTileCoords = GetEdgeTile(mapSize, rotation, EdgeType::RIGHT, false);
-    auto bottomTileCoords = GetEdgeTile(mapSize, rotation, EdgeType::BOTTOM, false);
+    const TileCoordsXY cornerCoords[2][4] = {
+        {
+            // Map corners
+            { 1, 1 },
+            { gMapSize.x - 2, gMapSize.y - 2 },
+            { 1, gMapSize.y - 2 },
+            { gMapSize.x - 2, 1 },
+        },
+        {
+            // Horizontal view clipping corners
+            TileCoordsXY{ CoordsXY{ std::max(gClipSelectionA.x, 32), std::max(gClipSelectionA.y, 32) } },
+            TileCoordsXY{ CoordsXY{ std::min(gClipSelectionB.x, (gMapSize.x - 2) * 32),
+                                    std::min(gClipSelectionB.y, (gMapSize.y - 2) * 32) } },
+            TileCoordsXY{ CoordsXY{ std::max(gClipSelectionA.x, 32), std::min(gClipSelectionB.y, (gMapSize.y - 2) * 32) } },
+            TileCoordsXY{ CoordsXY{ std::min(gClipSelectionB.x, (gMapSize.x - 2) * 32), std::max(gClipSelectionA.y, 32) } },
+        },
+    };
 
-    // Centre the coordinates so we don't have a hard crop at the edge of the visible tile
-    leftTileCoords += CoordsXY(16, 16);
-    rightTileCoords += CoordsXY(16, 16);
-    bottomTileCoords += CoordsXY(16, 16);
+    auto* const mainWindow = window_get_main();
+    const auto* const mainViewport = window_get_viewport(mainWindow);
+    const bool useViewClipping = (mainViewport != nullptr && mainViewport->flags & VIEWPORT_FLAG_CLIP_VIEW);
 
     // Calculate the viewport bounds
-    int32_t left = translate_3d_to_2d_with_z(rotation, CoordsXYZ(leftTileCoords, 0)).x;
-    int32_t top = GetTallestVisibleTileTop(mapSize, rotation);
-    int32_t right = translate_3d_to_2d_with_z(rotation, CoordsXYZ(rightTileCoords, 0)).x;
-    int32_t bottom = translate_3d_to_2d_with_z(rotation, CoordsXYZ(bottomTileCoords, 0)).y;
+    auto corners = cornerCoords[useViewClipping ? 1 : 0];
+    auto screenCoords1 = translate_3d_to_2d_with_z(rotation, { corners[0].ToCoordsXY().ToTileCentre(), 0 });
+    auto screenCoords2 = translate_3d_to_2d_with_z(rotation, { corners[1].ToCoordsXY().ToTileCentre(), 0 });
+    auto screenCoords3 = translate_3d_to_2d_with_z(rotation, { corners[2].ToCoordsXY().ToTileCentre(), 0 });
+    auto screenCoords4 = translate_3d_to_2d_with_z(rotation, { corners[3].ToCoordsXY().ToTileCentre(), 0 });
+
+    auto left = std::min({ screenCoords1.x, screenCoords2.x, screenCoords3.x, screenCoords4.x }) - 32;
+    auto top = GetTallestVisibleTileTop(rotation, corners[0], corners[1], useViewClipping);
+    auto bottom = std::max({ screenCoords1.y, screenCoords2.y, screenCoords3.y, screenCoords4.y });
```

Comment:
I was wondering why this doesn't need any subtraction since the coordinate is the center of the lowest tile, but that seems to be because the tiles at the map's edge have a base eight of 2 by default. This height offset just happens to be the same.